extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/videos/GoogleVideosInterface.java

```diff
@@ -0,0 +1,160 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.datatransfer.google.videos;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.*;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.http.json.JsonHttpContent;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.ArrayMap;
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.CharStreams;
+import org.datatransferproject.datatransfer.google.mediaModels.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class GoogleVideosInterface {
+  private static final int ALBUM_PAGE_SIZE = 20; // TODO
+  private static final int MEDIA_PAGE_SIZE = 50; // TODO
+
+  private static final String PAGE_SIZE_KEY = "pageSize";
+  private static final String TOKEN_KEY = "pageToken";
+  private static final String ALBUM_ID_KEY = "albumId";
+  private static final String ACCESS_TOKEN_KEY = "access_token";
+  private static final String FILTERS_KEY = "filters";
+  private static final String INCLUDE_ARCHIVED_KEY = "includeArchivedMedia";
+  private static final String MEDIA_FILTER_KEY = "mediaTypeFilter";
+  private static final String BASE_URL = "https://photoslibrary.googleapis.com/v1/";
+
+  private final ObjectMapper objectMapper =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+  private final HttpTransport httpTransport = new NetHttpTransport();
+  private final Credential credential;
+
+  GoogleVideosInterface(Credential credential) {
+    this.credential = credential;
+  }
+
+  String uploadVideoContent(InputStream inputStream, String filename) throws IOException {
+    InputStreamContent content = new InputStreamContent(null, inputStream);
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    content.writeTo(outputStream);
+    byte[] contentBytes = outputStream.toByteArray();
+    HttpContent httpContent = new ByteArrayContent(null, contentBytes);
+
+    return makePostRequest(BASE_URL + "uploads/", Optional.empty(), httpContent, String.class);
+  }
+
+  NewMediaItemResult createVideo(NewMediaItemUpload newMediaItemUpload) throws IOException {
+    HashMap<String, Object> map = createJsonMap(newMediaItemUpload);
+    HttpContent httpContent = new JsonHttpContent(new JacksonFactory(), map);
```

Comment:
Can we pass the jsonFactory in the constructor. I believe the transfer extension has it available form the context.

Reply:
Yup, reusing the JsonFactory now